### PR TITLE
ci(workflow): move 'generate_cli_screenshots' steps to 'docspublish'

### PR DESCRIPTION
**Pull request by marcosdotme**
Since the purpose is to updated the CLI screenshots on Github Pages, combining this two workflows makes sense.

Closes: #1125

## Checklist

- [ ] Add test cases to all the changes you introduce **_(not applicable)_**
- [ ] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test **_(not applicable)_**
- [x] Test the changes on the local machine manually
- [ ] Update the documentation for the changes **_(not applicable)_**

## Expected behavior
After every push to master branch, this Github Action Workflow will be triggered and will update the CLI screenshots and also updated the images on Github Pages (if any change is detected).

## Steps to Test This Pull Request
I tested manually using an clone of **commitizen** repo.

General view of workflow:

image

Update CLI screenshots using the scripts/gen_cli_help_screenshots.py (I add **ci_v3** to command descriptions just to workflow detect changes and to maintain traceability):

image

Update the images on Github Pages:

image

Image successfully updated on Github Pages:

image

## Relevant changes (@Lee-W)
- Renamed the workflow `generate-cli-screenshots` to `update-cli-screenshots`
- Added `[skip ci]` in the description of the commit generated by the job `update-cli-screenshots`. The final step of this job is create an commit and push to master. And the workflow is triggered by pushes to master, so if we not add `[skip ci]`, the workflow starts an infinite loop
- Add step "Pull latest changes" to capture the commit maded by `update-cli-screenshots`

### CODE REVIEW

**Comment by marcosdotme** [comment]
@Lee-W This PR refers to requested changes on #1122

File: .github/workflows/docspublish.yml
Comment (Lee-W) [inline .github/workflows/docspublish.yml:48]: nitpick: we probably would want to unify the style. (i.e., whether to add a blank line between steps or not)

**Review by Lee-W — APPROVED** [review]
Mostly good. Left a minor nitpick.

File: .github/workflows/docspublish.yml
Comment (marcosdotme) [reply .github/workflows/docspublish.yml:48]: I like to add blank line between steps but since `docspublish.yml` wasnt maded by me I dont like to change without permission. So we follow the blank line after steps?

Comment (Lee-W) [reply .github/workflows/docspublish.yml:48]: Just checked, looks like our convention is not to have blank lines so let's remove it.

Comment (Lee-W) [reply .github/workflows/docspublish.yml:48]: Also, it's perfectly fine to change others' code 🙂  If you have doubt, you can leave a comment or make those changes in a single commit in case we decide to revert it

Comment (marcosdotme) [reply .github/workflows/docspublish.yml:44]: Got it! Thanks!

Comment (marcosdotme) [reply .github/workflows/docspublish.yml:44]: I removed the blank lines an ammend it. Waiting for other maintainers to review 😄

Comment (Lee-W) [reply .github/workflows/docspublish.yml:44]: It's not functionality-related. I'll just merge this one once CI passes

Comment (Lee-W) [reply .github/workflows/docspublish.yml:44]: @marcosdotme I think we probably need to add *.svg to pre-commit `trim trailing whitespace` exceptions

Comment (marcosdotme) [reply .github/workflows/docspublish.yml:44]: @Lee-W CI passes first time, but when I remove the blank lines and ammend it, it fails 🙃. Trying to figure out why...

File: .github/workflows/docspublish.yml
Comment (Lee-W) [inline .github/workflows/docspublish.yml:33]:
```suggestion
          git commit -m "docs(cli/screenshots): update CLI screenshots" -m "[skip ci]"
```

what is this second -m for?

Comment (marcosdotme) [reply .github/workflows/docspublish.yml:33]: Is an git syntax for declaring the commit description. First "-m" is for commit title and second "-m" for commit description like this:

image

I don't like the "[skip ci]" on commit title so I put it on commit description.

Comment (Lee-W) [reply .github/workflows/docspublish.yml:33]: Got it 

```

       -m <msg>, --message=<msg>
           Use the given <msg> as the commit message. If multiple -m options are given, their values are concatenated as separate paragraphs.

           The -m option is mutually exclusive with -c, -C, and -F.

```

**Review by Lee-W — APPROVED** [review]

**Pull request merged**
